### PR TITLE
doc: translate hooks-rules

### PR DESCRIPTION
**Pull request by KevinHu2014**
Ready for first time review.

### CODE REVIEW

File: content/docs/hooks-rules.md [hunk 1]
```diff
@@ -76,63 +76,63 @@ function Form() {
...
-React wouldn't know what to return for the second `useState` Hook call. React expected that the second Hook call in this component corresponds to the `persistForm` effect, just like during the previous render, but it doesn't anymore. From that point, every next Hook call after the one we skipped would also shift by one, leading to bugs.
+React 不會知道第二個 `useState` Hook 呼叫要 return 什麼。 React 預期在這個 conponent 中的第二個 Hook 呼叫和 `persistForm` effect 是相對應的，就如同在前一次的 render 一樣，但它不再一樣了。 從那時起，在我們跳過的那個 Hook 後面，每下一個 Hook 呼叫都會 shift 一個，導致 bug 的發生。
 
-**This is why Hooks must be called on the top level of our components.** If we want to run an effect conditionally, we can put that condition *inside* our Hook:
+**這就是為何必須在我們的 component 之上層來呼叫 Hook。** 如果我們想要有條件地運行 effect ，我們可以把那個 condition 放在我們的 Hook *裡*：
 
 ```js
   useEffect(function persistForm() {
-    // 👍 We're not breaking the first rule anymore
+    // 👍 我們不再違反第一個規則
```
[the remark this answers is not in the document]
Comment (KevinHu2014) [reply content/docs/hooks-rules.md]: Hi @neighborhood999 thanks for the review.
I'm a bit confuse about whether the comment should be translated or not.
Cause I see on the first two section of the Hook doc, comments are being translated.
see as the links below
https://zh-hant.reactjs.org/docs/hooks-intro.html
https://zh-hant.reactjs.org/docs/hooks-overview.html

[the remark this answers is not in the document]
Comment (neighborhood999) [reply content/docs/hooks-rules.md]: Hi @KevinHu2014, I forgot how did I discuss this problem with @chloewlin in somewhere.

Sorry for making you confused, so keep your translation comments in here.

File: content/docs/hooks-rules.md [hunk 2]
```diff
@@ -1,73 +1,73 @@
...
 permalink: docs/hooks-rules.html
 next: hooks-custom.html
 prev: hooks-effect.html
 ---
 
-*Hooks* are a new addition in React 16.8. They let you use state and other React features without writing a class.
+*Hook* 是 React 16.8 新加的功能，他們讓你可以不用寫 class 就能使用 state 與其他 React 的功能。
 
-Hooks are JavaScript functions, but you need to follow two rules when using them. We provide a [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to enforce these rules automatically:
+Hook 是 JavaScript function，但你在使用他們時需要遵守兩個規則。 我們提供了一個 [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) 來自動化地實行這些規則：
```
[the remark this answers is not in the document]
Comment (KevinHu2014) [reply content/docs/hooks-rules.md]: There's a missing character c in react

**Comment by KevinHu2014** [comment]
Hi @neighborhood999 ,
I have apply the suggestions and sorry for separating those changes in multiple commits.
Ready for second time review.

File: content/docs/hooks-rules.md [hunk 3]
```diff
@@ -1,73 +1,73 @@
...
 permalink: docs/hooks-rules.html
 next: hooks-custom.html
 prev: hooks-effect.html
 ---
 
-*Hooks* are a new addition in React 16.8. They let you use state and other React features without writing a class.
+*Hook* 是 React 16.8 新加入的功能，它們讓你可以不用寫 class 就能使用 state 與其他 React 的功能。
 
-Hooks are JavaScript functions, but you need to follow two rules when using them. We provide a [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to enforce these rules automatically:
+Hook 是 JavaScript function，當你使用它們時需要遵守兩個規則。 我們提供了一個 [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) 來自動化地實行這些規則：
```
Comment (neighborhood999) [inline content/docs/hooks-rules.md]:
```suggestion
Hook 是 JavaScript function，當你使用它們時需要遵守兩個規則。我們提供了一個 [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) 來自動化地實行這些規則：
```

File: content/docs/hooks-rules.md [hunk 4]
```diff
@@ -1,73 +1,73 @@
...
+    "react-hooks/exhaustive-deps": "warn" // 檢查 effect 的相依性
   }
 }
 ```
 
-In the future, we intend to include this plugin by default into Create React App and similar toolkits.
+在未來，我們打算在 Create React App 和相關的 toolkit 中將這個套件設為預設。
 
-**You can skip to the next page explaining how to write [your own Hooks](/docs/hooks-custom.html) now.** On this page, we'll continue by explaining the reasoning behind these rules.
+**你現在可以先跳過，下一頁將解釋如何撰寫[你自己的 Hooks](/docs/hooks-custom.html)。** 在這頁，我們將會繼續解釋這些規則背後的原因。
```
Comment (neighborhood999) [inline content/docs/hooks-rules.md]:
```suggestion
**你現在可以先跳過，下一頁將解釋如何打造[你自己的 Hook](/docs/hooks-custom.html)。**在這頁，我們將會繼續解釋這些規則背後的原因。
```

File: content/docs/hooks-rules.md [hunk 5]
```diff
@@ -1,73 +1,73 @@
...
   useEffect(function persistForm() {
     localStorage.setItem('formData', name);
   });
 
-  // 3. Use the surname state variable
+  // 3. 使用 surname state 變數
   const [surname, setSurname] = useState('Poppins');
 
-  // 4. Use an effect for updating the title
+  // 4. 使用一個 effect 來更新 title
```
Comment (neighborhood999) [inline content/docs/hooks-rules.md]:
```suggestion
  // 4. 使用一個 effect 來更新標題
```

File: content/docs/hooks-rules.md [hunk 6]
```diff
@@ -76,63 +76,63 @@ function Form() {
...
+// 第一次 render
 // ------------
-useState('Mary')           // 1. Initialize the name state variable with 'Mary'
-useEffect(persistForm)     // 2. Add an effect for persisting the form
-useState('Poppins')        // 3. Initialize the surname state variable with 'Poppins'
-useEffect(updateTitle)     // 4. Add an effect for updating the title
+useState('Mary')           // 1. 用 'Mary' 來初始化 name state 變數 
+useEffect(persistForm)     // 2. 增加一個 effect 來保存表單
+useState('Poppins')        // 3. 用 'Poppins' 來初始化 surname state 變數
+useEffect(updateTitle)     // 4. 增加一個 effect 來更新 title
```
Comment (neighborhood999) [inline content/docs/hooks-rules.md]:
```suggestion
useEffect(updateTitle)     // 4. 增加一個 effect 來更新標題
```

File: content/docs/hooks-rules.md [hunk 7]
```diff
@@ -76,63 +76,63 @@ function Form() {
...
+// 第二次 render
 // -------------
-useState('Mary')           // 1. Read the name state variable (argument is ignored)
-useEffect(persistForm)     // 2. Replace the effect for persisting the form
-useState('Poppins')        // 3. Read the surname state variable (argument is ignored)
-useEffect(updateTitle)     // 4. Replace the effect for updating the title
+useState('Mary')           // 1. 讀取 name state 變數 (參數被忽略了)
+useEffect(persistForm)     // 2. 替換了用來保存表單的 effect
+useState('Poppins')        // 3. 讀取 surname state 變數 (參數被忽略了)
+useEffect(updateTitle)     // 4. 替換了用來更新 title 的 effect
```
Comment (neighborhood999) [inline content/docs/hooks-rules.md]:
```suggestion
useEffect(updateTitle)     // 4. 替換了用來更新標題的 effect
```

File: content/docs/hooks-rules.md [hunk 8]
```diff
@@ -76,63 +76,63 @@ function Form() {
...
 
 // ...
 ```
 
-As long as the order of the Hook calls is the same between renders, React can associate some local state with each of them. But what happens if we put a Hook call (for example, the `persistForm` effect) inside a condition?
+只要 Hook 在 render 時被呼叫的順序是一致的，React 可以將一些 local state 和它們一一聯繫在一起。但如果我們把一個 Hook 呼叫（例如，`persistForm` effect）放在條件式中會發生什麼事呢？
 
 ```js
-  // 🔴 We're breaking the first rule by using a Hook in a condition
+  // 🔴 我們違反了第一個規則，在 condition 中使用 Hook
```
Comment (neighborhood999) [inline content/docs/hooks-rules.md]:
```suggestion
  // 🔴 我們違反了第一個規則，在條件式中使用 Hook
```

**Pull request merged**